### PR TITLE
The Wrath of Indonesia: JavaX -> Jakarta Migration

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Get on latest version of conjure-java-runtime

**Implementation Description (bullets)**:
- Upgrade CJR versions.
- But this has transitives that break: Tritium and Jersey classes
- And this has further transitives: Dropwizard and its role in testing
- And that has a transitive effect: WireMock
- This PR resolves all of the aforementioned transitives. Thankfully the WireMock folks prepped a fat jar.
- I validated that we don't have cases where *both* JavaX and Jakarta things live on the classpath.

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Existing tests, for general integrity of library stack
- I used baseline-class-uniqueness to validate that no JavaX and Jakarta things simultaneously live on the classpath. I did not leave it there as there were unrelated issues between AssertJ and ByteBuddy that I don't have the bandwidth to resolve right now.

**Concerns (what feedback would you like?)**:
- While the tests work, I still have a feeling that some of the HTTP stack stuff is risky, especially since we bumped Dropwizard through two major versions. The scary dependency-resolution problems I had would not affect prod code, so I'm a bit less worried, but that's worth validating.

**Where should we start reviewing?**: versions.props

**Priority (whenever / two weeks / yesterday)**: this week, if not in by tomorrow we should probably go with #4655 first

### CODE REVIEW

**Comment by iamdanfox** [comment]
Is there a plan to get onto GCV? having such a long list of forces in versions.props is a little scary :/

**Comment by jkozlowski** [comment]
I think we should pull this into an internal product to just sanity check it starts up alright?

File: versions.props [hunk 1]
```diff
@@ -43,13 +44,17 @@ io.airlift:airline = 0.8
 io.atomix.catalyst:* = 1.1.2
 io.atomix.copycat:* = 1.2.0
 io.atomix:* = 1.0.0-rc9
-io.dropwizard.metrics:metrics-* = 3.2.3
+io.dropwizard.metrics:metrics-* = 4.1.5
```
Comment (jkozlowski) [inline versions.props]: Will this not effectively cause a bump in version for anyone that consumes atlasdb?

Comment (jeremyk-91) [reply versions.props]: Yes, it will. I'll see if we can separate the test and compile classpaths.

Comment (j-baker) [reply versions.props]: we can't be on dropwizard metrics 4

Comment (j-baker) [reply versions.props]: it has enough breaking changes here

Comment (jeremyk-91) [reply versions.props]: Now we're not. I went back to metrics 3

**Comment by jeremyk-91** [comment]
Validated on internal ski product in deployment to an internal stack. Pressing the big red button ™️

**Comment by jeremyk-91** [comment]
GCV: I'll open a task for tracking, it's probably worth doing soon. I think it's fair to say someone will look at this in the next 1-2 weeks, though I don't know how deep that rabbithole goes, so I can't commit the team to doing it in that time span.

**Pull request merged**